### PR TITLE
solution

### PR DESCRIPTION
**Pull request by MaxSchmide**
[DEMO LINK](https://maxschmide.github.io/react_movies-list-filter-js/)

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,33 +1,53 @@
...
+              Search movie
+            </label>
+
+            <div className="control">
+              <input
+                type="text"
+                id="search-query"
+                className="input"
+                placeholder="Type search word"
+                onChange={e => setQuery(e.target.value)}
```
Comment (irinabolbat) [inline src/App.jsx:39]: you should add atribute `value={query}`

**Review by irinabolbat — APPROVED** [review]
good job 👍 
please, check my comment

File: src/App.jsx (hunk 1 above)
Comment (anstsot) [reply src/App.jsx:39]: agree with comment

File: src/App.jsx [hunk 2]
```diff
@@ -1,33 +1,54 @@
...
-              type="text"
-              id="search-query"
-              className="input"
-              placeholder="Type search word"
-            />
+const filterMoviesByWord = (movies, query) => {
+  const filtered = movies.filter((movie) => {
+    const title = movie.title.toLowerCase();
+    const desc = movie.description.toLowerCase();
+    const q = query.toLowerCase().trim();
```
Comment (anstsot) [inline src/App.jsx:10]: please rename to the `normalizedQuery` and do not use one-letter variable namings

File: src/App.jsx [hunk 3]
```diff
@@ -1,33 +1,54 @@
...
-
-          <div className="control">
-            <input
-              type="text"
-              id="search-query"
-              className="input"
-              placeholder="Type search word"
-            />
+const filterMoviesByWord = (movies, query) => {
+  const filtered = movies.filter((movie) => {
```
Comment (anstsot) [inline src/App.jsx:7]: you can immediately return instead of creating additional variable
```suggestion
  return movies.filter((movie) => {
```

**Review by anstsot — CHANGES_REQUESTED** [review]
ood start:four_leaf_clover:
Let’s add a few fixes to improve your code.

File: src/App.jsx [hunk 4]
```diff
@@ -1,33 +1,51 @@
...
-              type="text"
-              id="search-query"
-              className="input"
-              placeholder="Type search word"
-            />
+const filterMoviesByWord = (movies, query) => {
+  return movies.filter((movie) => {
+    const title = movie.title.toLowerCase();
+    const desc = movie.description.toLowerCase();
+    const normalizedQuery = query.toLowerCase().trim();
```
Comment (qoa11a) [inline src/App.jsx:10]: It's better to move it outside of the filter method not to do it on every iteration.

**Review by qoa11a — APPROVED** [review]

**Pull request closed without merging**